R/tmb.R

```diff
@@ -588,16 +587,20 @@ h_mmrm_tmb <- function(formula,
     silent = TRUE
   )
   h_mmrm_tmb_assert_start(tmb_object)
-  tmb_opt <- with(
+  args <- with(
     tmb_object,
-    do.call(
-      what = control$optimizer,
-      args = c(
-        list(par, fn, gr, control = control$optimizer_control),
-        control$optimizer_args
-      )
+    c(
+      list(par, fn, gr, hessian = he, control = control$optimizer_control),
+      control$optimizer_args
     )
   )
+  if (!identical(control$optimizer, stats::nlminb)) {
```

Comment:
Is this a stable way of checking optimizer class? Probably, just wanted to double check.

Reply:
`identical` should work in checking the functions if they are identical here

Reply:
but we can save the `optimizer` in character and save it to `mmrm_control`

Reply:
hm I guess better compare the functions, as is

Reply:
sorry a minor thing here to improve, I would do it like first setting the args without the hessian, and then if we are in the special case of nlminb add the hessian